src/monodroid/jni/inmemory-assemblies.cc

```diff
@@ -0,0 +1,154 @@
+#include "inmemory-assemblies.hh"
+#include "globals.hh"
+
+#include <mono/metadata/appdomain.h>
+#include <mono/metadata/assembly.h>
+#include <mono/metadata/reflection.h>
+
+extern "C" {
+#include "java-interop-util.h"
+}
+
+namespace xamarin::android::internal {
+
+void
+InMemoryAssemblies::add_or_update_from_java (MonoDomain *domain, JNIEnv *env, jstring_array_wrapper &assemblies, jobjectArray assembliesBytes)
+{
+	if (assembliesBytes == nullptr)
+		return;
+
+	unsigned int len = static_cast<unsigned int> (env->GetArrayLength (assembliesBytes));
+	char **names = new char*[len];
+	char **assemblies_bytes = new char*[len];
+	unsigned int *assemblies_bytes_len = new unsigned int[len];
+
+	for (unsigned int index = 0; index < len; index++) {
```

Comment:
I half-wonder if this shouldn't be moved into an `InMemoryAssemblyEntry` constructor...

Reply:
Now that it has a constructor I would say that's a good point, I can do that